src/components/Transactions.js

```diff
@@ -0,0 +1,94 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import axios from 'axios';
+
+export default class Transactions extends React.Component {
+  constructor() {
+    super();
+    this.state = {
+      transactions: [],
+      loadingSuccess: null,
+      errorMessage: '',
+    };
+  }
+
+  async componentDidMount() {
+    try {
+      const { data } = await axios.get('/api/transactions');
+      const transactions = [];
+      const existsTest = {};
+      while (data.length) {
+        // Have fun figuring out what the heck I'm doing here :)
```

Comment:
I see what you are doing, and i like it.

Reply:
Nice man. I felt this may have been a little convoluted.
This one was a serious win for me personally because it was one of those magic moments where it just worked the first time. There was no debugging to be done as far as I can see.